velox/expression/CastExpr.cpp

```diff
@@ -151,35 +176,12 @@ VectorPtr castToDate(
       const auto& queryConfig = context.execCtx()->queryCtx()->queryConfig();
       auto sessionTzName = queryConfig.sessionTimezone();
       if (queryConfig.adjustTimestampToTimezone() && !sessionTzName.empty()) {
-        // locate_zone throws runtime_error if the timezone couldn't be found
-        // (so we're safe to dereference the pointer).
         auto* timeZone = date::locate_zone(sessionTzName);
-        context.applyToSelectedNoThrow(rows, [&](int row) {
-          auto input = inputVector->valueAt(row);
-          input.toTimezone(*timeZone);
-          auto seconds = input.getSeconds();
-          if (seconds >= 0 || seconds % kSecsPerDay == 0) {
-            resultFlatVector->set(row, seconds / kSecsPerDay);
-          } else {
-            // For division with negatives, minus 1 to compensate the discarded
-            // fractional part. e.g. -1/86'400 yields 0, yet it should be
-            // considered as -1 day.
-            resultFlatVector->set(row, seconds / kSecsPerDay - 1);
-          }
-        });
+        castTimestampToDate<true>(
```

Comment:
It might be cleaner to pass const BaseVector& input as a parameter to the callTimestampToDate function and add the line
 auto* inputVector = input.as<SimpleVector<Timestamp>>(); 

as the first line in its function body.

Reply:
Done.